### PR TITLE
Migrate CI/CD to Github Action

### PR DESCRIPTION
**Pull request by exelban**
Hi.

I take all jobs from circle CI and migrate them to github actions.
I add additional action which builds a docker image and deploys it to docker hub when a new tag is created (#61).

You need to add a few secrets to ensure access to codecov and docker hub.
`CODECOV_TOKEN` - codecov token
`DOCKER_USERNAME` - docker username
`DOCKER_PASSWORD` - docker password (but I propose to use [docker token](https://docs.docker.com/docker-hub/access-tokens/))

### CODE REVIEW

**Comment by aperezg** [comment]
wow that so fast! thanks! 

could you change the docker login to use the token(`DOCKER_TOKEN`) instead user and password like you propose ? :D

**Comment by exelban** [comment]
You can see this workflows in my [fork](https://github.com/exelban/killgrave/actions?query=workflow%3ABuild). Coverage workflow is falling because there is no CODECOV.

File: .github/workflows/release.yaml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+      - name: Login to docker hub
+        uses: actions-hub/docker/login@master
+        env:
+          DOCKER_USERNAME: ${{ secrets.DOCKER_USERNAME }}
+          DOCKER_PASSWORD: ${{ secrets.DOCKER_TOKEN }}
+
+      - name: Build image
+        if: success()
+        run: docker build -t ${GITHUB_REPOSITORY}/cli:${IMAGE_TAG} .
```
Comment (aperezg) [inline .github/workflows/release.yaml:35]:
```suggestion
        run: docker build -t ${GITHUB_REPOSITORY}:${IMAGE_TAG} .
```

File: .github/workflows/release.yaml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+      - name: Push to docker registry
+        if: success()
+        uses: actions-hub/docker/cli@master
+        with:
+          args: push ${GITHUB_REPOSITORY}/cli:${IMAGE_TAG}
```
Comment (aperezg) [inline .github/workflows/release.yaml:41]:
```suggestion
          args: push ${GITHUB_REPOSITORY}:${IMAGE_TAG}
```

File: .github/workflows/release.yaml (hunk 1 above)
Comment (aperezg) [inline .github/workflows/release.yaml:35]: If I not wrong you need to declare the $IMAGE_TAG variable, isn't it?

**Review by aperezg — CHANGES_REQUESTED** [review]

File: .github/workflows/release.yaml (hunk 1 above)
Comment (exelban) [reply .github/workflows/release.yaml:35]: Yeap, it looks like I forgot it.

Comment (exelban) [reply .github/workflows/release.yaml]: Nope, I forgot again. This [docker cli](https://github.com/actions-hub/docker/blob/master/cli/entrypoint.sh) create an env variable called IMAGE_TAG and IMAGE_NAME (which are GITHUB_REPOSITORY). So this env is already there.

**Comment by exelban** [comment]
I can also add a step to deploy the image as `latest`. If it's needed.

**Comment by aperezg** [comment]
> I can also add a step to deploy the image as `latest`. If it's needed.

yes that will be nice

**Review by aperezg — APPROVED** [review]

**Comment by aperezg** [comment]
@joanlopez take a look, thanks :D

**Review by joanlopez — APPROVED** [review]
LGTM!! Great job 👍

cc/ @aperezg

**Pull request merged**